app/controllers/orders_controller.rb

```diff
@@ -316,7 +316,8 @@ def grouped_order_details
     # Do not do any additional scopes on `@order.order_details` to ensure the list
     # of OrderDetail objects is memoized. Validations are run on `@order.order_details`
     # and we want to be able to render validation errors inline.
-    @order.order_details.sort_by(&:group_id).slice_when do |before, after|
+    sorted_order_details = @order.order_details.sort_by { |od| od.group_id.to_i }
```

Comment:
Actually, I think it might make sense to pull the whole thing up into a model method on `Order`. 'm not sure giving it a default value makes sense. In this value, I do think `null` is more meaningful than `0`.  I like the `safe_group_id` idea.